File: backend/data/projects/392a52dd-119c-46c9-9513-726e5066c289/code/frontend/src/types/messages.ts
import { UUID } from './common';

export interface MessageBase {
  content: string;
}

export interface MessageCreate extends MessageBase {
  receiver_id: UUID;
}

export interface Message extends MessageBase {
  message_id: UUID;
  conversation_id: UUID;
  sender_id: UUID;
  receiver_id: UUID;
  sent_at: string;
  read_at: string | null;
}

export interface ConversationBase {
  parent_id: UUID;
  child_id: UUID;
}

export interface ConversationCreate extends ConversationBase {}

export interface Conversation extends ConversationBase {
  conversation_id: UUID;
  created_at: string;
  updated_at: string;
}

export interface ConversationWithMessages extends Conversation {
  messages: Message[];
}

File: backend/data/projects/392a52dd-119c-46c9-9513-726e5066c289/code/backend/src/schemas/messages.py
from pydantic import BaseModel, Field
import uuid
from datetime import datetime
from typing import Optional, List


class MessageBase(BaseModel):
    content: str = Field(..., example="Hi sweetie! Don't forget your soccer practice at 5 PM.")


class MessageCreate(MessageBase):
    receiver_id: uuid.UUID


class Message(MessageBase):
    message_id: uuid.UUID
    conversation_id: uuid.UUID
    sender_id: uuid.UUID
    receiver_id: uuid.UUID
    sent_at: datetime
    read_at: Optional[datetime] = None

    class Config:
        orm_mode = True


class ConversationBase(BaseModel):
    parent_id: uuid.UUID
    child_id: uuid.UUID


class ConversationCreate(ConversationBase):
    pass


class Conversation(ConversationBase):
    conversation_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True


class ConversationWithMessages(Conversation):
    messages: List[Message] = []

